### PR TITLE
Begin adding cooking proficiency infrastructure

### PR DESCRIPTION
**Pull request by I-am-Erk**
#### Summary
Infrastructure "Adds definitions for cooking proficiencies in JSON"

#### Purpose of change
See https://github.com/CleverRaven/Cataclysm-DDA/discussions/49191 for some details
This adds implementation for those cooking proficiencies which could reasonably have an effect on our current boolean fail state, or which should have a noticeable impact on crafting time.

#### Describe the solution
```
Food Preparation - adds a small time boost to most recipes
 -- Knife Skills - adds a large time boost to some recipes
Principles of Baking
 -- Breadmaking
 -- Dessert Baking - this one should be quite easy to learn but stems off into more advanced things.
  -- Advanced Dessert Baking
  -- Fried Desserts - also requires Principles of Frying
Principles of Frying - does not add as much of a fail multiplier as most "principles" do
 -- Breaded & Battered
 -- Fried Desserts - also requires Dessert Baking
Principles of Cheesemaking
 -- Cheesemaking Expert
Principles of Distillation (this will be used in more areas than just cooking)
 -- Beverage Distillation
Culinary Fermentation
 -- Brewing
 -- Winemaking
 -- Lactofermenting
Food Preservation
 -- Pressure Canning
 -- Food Curing/Drying
Principles of Cheesemaking
 -- Cheesemaking Expert
Scavenger Cooking - this may not work with current system but would be useful for recipes that people are unlikely to have made prior to the apocalypse
Forager Cooking - eg Pemmican, wastebread
```
Adds most of these proficiencies to cookbooks already in game.

#### Describe alternatives you've considered

Not all these proficiencies need to be set up exactly how I've decided to do it. I'm open to suggestions.

There are several other cooking proficiencies not included here, and I'm sure I'll think of more as I go. In particular things related to, for example, milling grains and such seem like they'd have a place here.

I'm leaving out the ones that shouldn't affect the current fail state for now: if it would just make the food taste bad, but not be inedible, to fail, then the proficiency should probably wait for now.

#### Testing

Not yet.

#### Additional context

These are not yet in recipes nor professions. The former would add too many lines, but I may go into the latter at this time so that when this is implemented not as many people will find their bakers inexplicably unable to bake.

### CODE REVIEW

File: data/json/proficiencies/comestibles.json [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+  },
+  {
+    "type": "proficiency",
+    "id": "prof_baking",
+    "name": { "str": "Principles of Baking" },
+    "description": "You know the basics of baking - proportions, leavening, and things - and are less likely to make a dramatic mistake when working from memory.",
+    "can_learn": true,
+    "default_time_multiplier": 1.5,
+    "default_fail_multiplier": 1.5,
+    "time_to_learn": "10 h",
```
Comment (LaVeyanFiend) [inline data/json/proficiencies/comestibles.json:31]:
```suggestion
    "time_to_learn": "10 h"
```

**Comment by actual-nh** [comment]
How about knowledge of vegan, lactose intolerant, etc substitutions - particularly ones that don't require pre-prepared (and otherwise hard to get after a short bit after the Cataclysm) stuff? (Side note - have to check into growing of nutritional yeast...)

**Comment by I-am-Erk** [comment]
A lot of what you're talking about would be good down the road when I can either specify certain proficiencies for substitutions, or make certain proficiencies needed to prevent mild fail states (the food tastes bad). I don't want to get too far into little specific proficiencies that if failed just make food taste less good until that outcome is possible.

**Comment by actual-nh** [comment]
> A lot of what you're talking about would be good down the road when I can either specify certain proficiencies for substitutions,

Agreed.

> or make certain proficiencies needed to prevent mild fail states (the food tastes bad).

Should having lactose or gluten contamination for a character with lactose or grain intolerance be considered a mild fail state?

**Comment by ashGlaw** [comment]
I love the idea of mild fail states. For one-use items like food this makes tons of sense

**Comment by PatrikLundell** [comment]
I think you should assume that anyone who survived the apocalypse having severe grain and/or lactose intolerance would know enough basics not to contaminate their food with the products they can't tolerate. If the player doesn't read the labels/know their ingredients tables and have their characters cook something up using harmful ingredients then the result is according to the recipe (more or less flawed, of course). You could argue that the game ought to help players to recognize what their characters would know (e.g. "are you sure you want to use lactose containing ingredients?", or color such ingredients differently), but that's really a different issue.

I don't really see why varying baking should be harder than varying other cooking. In both cases you're making one batch and probably won't make another one until you've consumed the previous one (or discarded it, if you managed to screw up that badly).

**Comment by I-am-Erk** [comment]
> I think you should assume that anyone who survived the apocalypse having severe grain and/or lactose intolerance would know enough basics not to contaminate their food with the products they can't tolerate. If the player doesn't read the labels/know their ingredients tables and have their characters cook something up using harmful ingredients then the result is according to the recipe (more or less flawed, of course). You could argue that the game ought to help players to recognize what their characters would know (e.g. "are you sure you want to use lactose containing ingredients?", or color such ingredients differently), but that's really a different issue.
> 
> I don't really see why varying baking should be harder than varying other cooking. In both cases you're making one batch and probably won't make another one until you've consumed the previous one (or discarded it, if you managed to screw up that badly).

This is all making the assumption a character is using a recipe. Recipe books will be proficiency books, with a fairly good mitigation of failure. Having the proficiency makes you more able to prepare recipes for things without the book.

For example, you can make bread pretty easily with a cookbook and a recipe. Then you memorize the recipe. Then you try to make bread again, but if you've not done much baking, you have a solid chance of failing when trying to do it from memory.

File: data/json/proficiencies/chemistry.json [hunk 2]
```diff
@@ -1,6 +1,5 @@
 [
   {
-    "type": "proficiency",
     "id": "prof_intro_chemistry",
```
Comment (kevingranade) [inline data/json/proficiencies/chemistry.json:3]:
```suggestion
    "type": "proficiency",
    "id": "prof_intro_chemistry",
```

**Pull request merged**